sabnzbd/misc.py

```diff
@@ -604,22 +605,32 @@ def to_units(val: Union[int, float], postfix="") -> str:
 
     if val < 0:
         sign = "-"
+        val = abs(val)
     else:
         sign = ""
 
     # Determine what form we are at
-    val = abs(val)
-    n = 0
-    while (val > 1023) and (n < 5):
-        val = val / 1024
-        n = n + 1
+    # Anything beyond petabytes is outside of scope
+    if val == 0:
+        n = 0
+    else:
+        n = min(5, math.trunc(math.log2(val) / 10))
+
+    val = val / 2 ** (10 * n)
```

Comment:
What's the formula you're using here?

And what speedup does it achieve? 
The previous code was really Pythonic code and easy to read and understand. It's a mantra that I would like to keep unless major speedups can be achieved.

Reply:
Thanks for the assist, @jcfp. You've described well what is happening.  It seems it's not as obvious as I thought, so I will add some commentary.

> the function is ever exposed to small enough numbers

I did consider this, but thought it out of the domain since these are seemingly always bytes.  (Thus the `1024` instead of `1000` for more SI-like units.) That being said, it does also sometimes operate on rates which might go very low.  Also, the fix is simple:  I am already mapping `0` to `0` above; I'll simply extend that to `1024`.